lib/src/model/game/game_filter.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:chessground/chessground.dart';
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:lichess_mobile/src/model/common/perf.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'game_filter.freezed.dart';
+part 'game_filter.g.dart';
+
+@riverpod
+class GameFilter extends _$GameFilter {
+  @override
+  GameFilterState build({GameFilterState? filter}) {
+    return filter ?? const GameFilterState();
+  }
+
+  void setFilter(GameFilterState filter) => state = state.copyWith(
+        perfs: filter.perfs,
+        side: filter.side,
+      );
+
+  int countFiltersInUse() {
```

Comment:
Provider should not define public getters or expose state through public methods.

The `count` info is already available in the state. A `count` getter can be defined in the `GameFilterState` class directly if needed.